compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -351,3 +403,26 @@ impl<'a> LlvmArchiveBuilder<'a> {
 fn string_to_io_error(s: String) -> io::Error {
     io::Error::new(io::ErrorKind::Other, format!("bad archive: {}", s))
 }
+
+fn find_dlltool(sess: &Session) -> OsString {
+    // When cross-compiling first try binary prefixed with target triple
+    if sess.host.llvm_target != sess.target.llvm_target {
```

Comment:
`llvm_target` is probably not the best way to check for this. There are many llvm target triples that are actually equivalent in the end. (e.g. `i686-pc-windows-gnu` and `i686-unknown-windows-gnu` and e.g. json targets could specify either).